management/daemon.py

```diff
@@ -121,8 +121,9 @@ def index():
 	no_users_exist = (len(get_mail_users(env)) == 0)
 	no_admins_exist = (len(get_admins(env)) == 0)
 
-	import boto.s3
-	backup_s3_hosts = [(r.name, r.endpoint) for r in boto.s3.regions()]
+	import boto3.s3
+	from urllib.parse import urlparse
+	backup_s3_hosts = [(r, urlparse(boto3.client('s3', region_name=r).meta.endpoint_url).netloc) for r in boto3.session.Session().get_available_regions('s3')]
```

Comment:
This is causing a very very slow page load for me. Since valid endpoint URLs are predictable from the region name, this should be better:

```suggestion
	backup_s3_hosts = [(r, f"s3.{r}.amazonaws.com") for r in boto3.session.Session().get_available_regions('s3')]
```

The endpoint URL for us-east-1 changes from `s3.amazonaws.com` to `s3.us-east-1.amazonaws.com` but this is valid according to https://docs.aws.amazon.com/general/latest/gr/s3.html.

The only concern would be if this breaks any existing backup settings using us-east-1, but I don't think so. This is only used for populating the drop-down list, and it seems like setting the drop-down to the current region is broken anyway (I'll fix this separately).

Reply:
Yeah, I will admit this was subobtimal. I like your workaround. its pushed.